pkg/lane/quic_test.go

```diff
@@ -0,0 +1,182 @@
+package lane
+
+import (
+	"errors"
+	"github.com/golang/mock/gomock"
+	"github.com/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/viaduct/pkg/packer"
+	"github.com/kubeedge/viaduct/pkg/translator"
+	"github.com/lucas-clemente/quic-go"
+	"reflect"
+	"testing"
+	"time"
+)
+
+var err1 error
+
+// TestNewQuicLane is function to test NewQuicLane().
+func TestNewQuicLane(t *testing.T) {
+	initMocks(t)
+	tests := []struct {
+		name string
+		van interface{}
+		want *QuicLane
+	}{
+		{
+			name:"NewQuickLaneTest-StreamObject",
+			van:mockStream,
+			want:&QuicLane{stream:mockStream},
+		},
+		{
+			name:"NewQuickLaneTest-EmptyObject",
+			van:"",
+			want:nil,
+		},
+	}
+	for _, tt := range tests {
+
+		t.Run(tt.name, func(t *testing.T) {
+			if got := NewQuicLane(tt.van); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("NewQuicLane() = %v, want %v", got, tt.want)
+			}
+
+		})
+	}
+}
+
+// TestQuicLane_ReadMessage is function to test ReadMessage().
+func TestQuicLane_ReadMessage(t *testing.T) {
+	var msg  = model.Message{Content:"message"}
+	bytesMsg, _ :=translator.NewTran().Encode(&msg)
+	header := packer.PackageHeader{Version:0011,PayloadLen:(uint32(len(bytesMsg)))}
+	headerBuffer := make([]byte, 0)
+	header.Pack(&headerBuffer)
+	err1=errors.New("Error")
+	tests := []struct {
+		name    string
+		stream        quic.Stream
+		msg *model.Message
+		wantErr error
+	}{
+		{
+			name:"Test-FailureCase",
+			stream:mockStream,
+			msg:&model.Message{},
+			wantErr:err1,
+		},
+		{
+			name:"Test-SuccessCase",
+			stream:mockStream,
+			msg:&model.Message{Content:"message"},
+			wantErr:nil,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			l := &QuicLane{
+				stream:        tt.stream,
+			}
+			if ( tt.name == "Test-FailureCase") {
```

Comment:
the main idea of table driven tests is easy addition of test. We have if else condition based on name of the test which should be unique for all tests, if we want to add 3rd we may require another else if. Please refactor.